homeassistant/helpers/entity_component.py

```diff
@@ -77,11 +98,10 @@ def async_setup(self, config):
 
         # Generic discovery listener for loading platform dynamically
         # Refer to: homeassistant.components.discovery.load_platform()
-        @callback
+        @asyncio.coroutine
         def component_platform_discovered(platform, info):
             """Handle the loading of a platform."""
-            self.hass.async_add_job(
-                self._async_setup_platform(platform, {}, info))
+            yield from self._async_setup_platform(platform, {}, info)
```

Comment:
So it look better but I think before it was faster. Ben mean that a coro they wait of a other coro without any reason is slower than only schedule a job.

Reply:
I think that we're micro-optimizing here and it is costing us meaningful stack traces.